stores/directory/local.py

```diff
@@ -19,46 +22,79 @@ def __init__(self, local_dir: Union[str, Path]):
             local_dir_path = local_dir
 
         # Make sure it exists and it's a directory.
-        assert local_dir_path.exists(), f"Given path {local_dir_path} does not exist."
-        assert local_dir_path.is_dir(), f"Given path {local_dir_path} is not a directory."
+        assert local_dir_path.exists(), f"Given path {
+            local_dir_path} does not exist."
+        assert local_dir_path.is_dir(), f"Given path {
+            local_dir_path} is not a directory."
 
         # Store that location against the class
-
         self.local_path = local_dir_path
 
-
     def add_file(self, file: Path):
         # Abstract methods can remain without implementation for now.
         pass
 
     def has_lone_file_matching(self, pattern: str) -> bool:
-        #  To be implemented in local directory store 1 #5 
-        ...
-
-    def save_lone_file_matching(self, pattern: str, save_as: str) -> Path:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Grab a list of files matching the regex pattern to determin how many exist.
+        matching_files = [
+            f for f in self.get_file_names if re.search(pattern, f)]
+
+        if matching_files.count == 1:  # 1 file matched
+            return True
+        elif matching_files.count == 0:  # 0 file matched
+            return False
+        elif matching_files.count > 1:  # 2+ files matched
+            raise ValueError(f"More than 1 file found that matches the regex pattern in gven directory {matching_files}") 
+
+    def save_lone_file_matching(self, pattern: str, destination: Optional[Union[Path, str]] = None):
+        # Assert 1 file matches
+        if self.has_lone_file_matching(pattern):
```

Comment:
this if needs an else (for if `has_lone_file_matching` == False)  that else needs to raise an error otherwise if people try and save but use a file pattern that doesn't match, nothing will happen but it will complete without any errors - would be confusing.